src/OpenTelemetry.Api/Trace/SpanHelper.cs

```diff
@@ -27,40 +27,26 @@ public static class SpanHelper
 #pragma warning disable CA1805 // Do not initialize unnecessarily
         private static readonly Status DefaultStatus = default;
 #pragma warning restore CA1805 // Do not initialize unnecessarily
-        private static readonly Dictionary<StatusCanonicalCode, string> StatusCanonicalCodeToStringCache = new Dictionary<StatusCanonicalCode, string>()
+        private static readonly Dictionary<StatusCode, string> StatusCodeToStringCache = new Dictionary<StatusCode, string>()
```

Comment:
Given there are only three values, probably no need to have this dictionary, use if/else directly?

Reply:
I think the idea here was to prevent the ToString in each call when we use this:
```csharp
        public static string GetCachedStatusCodeString(StatusCode statusCode)
        {
            if (!StatusCodeToStringCache.TryGetValue(statusCode, out string code))
            {
                return statusCode.ToString();
            }

            return code;
        }
```

Reply:
The string constants can be stored somewhere else, the key here is to avoid a dictionary look up since we only have 3 conditions.